client/src/model/slices/demographic_data_slice.ts

```diff
@@ -25,6 +29,11 @@ const demographicDataSlice = createSlice({
     getDemographicsError(state: DemographicDataState, action) {
       console.log(`Error fetching demographic data: ${state} ${action}`);
     },
+    demographicDataCleared(state: GeoDataState, action) {
```

Comment:
Could you try one thing both here and in other slicers. 
```
    getDemographicsSuccess(state: DemographicDataState, action: PayloadAction<DemographicData>) {
      return {
        ...state,
        data: { ...action.payload },
        status: { status: Status.pending },
      };
    },

...

export const clearDemographicData = () => {
  return async (dispatch: AppDispatch) => {
    dispatch(getDemographicsSuccess({}));
  };
};
```

Reply:
Oh that's a good point, will give that a try!

Reply:
As discussed offline will keep separate since we don't want to retain existing state info on clear.